app/services/claim_fast_tracking/max_rating_annotator.rb

```diff
@@ -64,10 +64,16 @@ def self.get_ratings(diagnostic_codes)
       nil
     end
 
-    def self.eligible_for_request?(dc)
-      Flipper.enabled?(:disability_526_maximum_rating_api_all_conditions) || SELECT_DISABILITIES.include?(dc)
+    def self.eligible_for_request?(rated_disability)
+      return false if %i[infectious_disease missing_diagnostic_code].include?(diagnostic_code_type(rated_disability))
+
+      return false if EXCLUDED_DIGESTIVE_CODES.include?(rated_disability.diagnostic_code)
+
+      return true if Flipper.enabled?(:disability_526_maximum_rating_api_all_conditions)
+
+      SELECT_DISABILITIES.include?(rated_disability.diagnostic_code)
     end
 
-    private_class_method :get_ratings, :eligible_for_request?
+    private_class_method :get_ratings
```

Comment:
`eligible_for_request?` (a no-side-effects utility method) is now publicly accessible, so we can write better / more comprehensive tests for it.